taf/UP/UP_Runner.py

```diff
@@ -274,6 +274,118 @@ def max_run_id(self, file="", tbl="", inyear=""):
+        """
+        Function create_efts_metadata to get the count of the given table by state and insert into the EFT
+        metadata table. Will be called in the get_segment Function.
+        """
+
+        #  Create state counts and insert into metadata table
+
+        z = f"""
+            create or replace temporary view record_count as
+            select count(*) as row_cnt
+            from {self.DA_SCHEMA}.TAF_ANN_UP_{tblname}
+            where da_run_id = {self.DA_RUN_ID}
+        """
+        self.append('UP', z)
+
+        z = f"""
+            create or replace temporary view state_counts as
+            select da_run_id,
+                submtg_state_cd,
+                count(submtg_state_cd) as rowcount_state
+
+            from {self.DA_SCHEMA}.TAF_ANN_UP_{tblname}
+            where da_run_id = {self.DA_RUN_ID}
+            group by da_run_id,
+                     submtg_state_cd
+        """
+        self.append('UP', z)
+
+        #    Insert the following values into the table:
+        #     - DA_RUN_ID
+        #     - 4th node text value
+        #     - table name
+        #     - year
+        #     - version number (01)
+        #     - overall table count
+        #     - run date
+        #     - state code
+        #     - table count for given state
+        #     - fil_dt
+        #     - version
+
+        z = f"""
+                INSERT INTO {self.DA_SCHEMA}.job_otpt_meta
+                SELECT {self.DA_RUN_ID} AS da_run_id
+                    ,'TABLE' AS otpt_type
+                    ,'TAF_ANN_UP_{tblname}' AS otpt_name
+                    ,'{self.DA_SCHEMA}' AS otpt_lctn_txt
+                    ,row_cnt as tot_rec_cnt
+                    ,'{fil_4th_node}' AS fil_4th_node_txt
+                    ,from_utc_timestamp(current_timestamp(), 'EST') as rec_add_ts
+                    ,NULL as rec_updt_ts
+                    ,NULL as trnct_ts
+                FROM record_count
+        """
+        self.append('UP', z)
+
+        z = f"""
+            insert into {self.DA_SCHEMA}.EFTS_FIL_META
+            (
+                 da_run_id
+                ,fil_4th_node_txt
+                ,otpt_name
+                ,rptg_prd
+                ,itrtn_num
+                ,tot_rec_cnt
+                ,fil_cret_dt
+                ,incldd_state_cd
+                ,rec_cnt_by_state_cd
+                ,rec_add_ts
+                ,rec_updt_ts
+                ,fil_dt
+                ,taf_cd_spec_vrsn_name
+                ,rfrsh_vw_flag
+                ,ltst_run_ind
+                ,ccb_qtr
+                ,rif_finl_vrsn
+                ,rif_prelim_vrsn
+            )
+            select
+                 a.da_run_id
+                ,'{fil_4th_node}' as fil_4th_node_txt
+                ,'TAF_ANN_UP_{tblname}' as otpt_name
+                ,'{self.year}' as rptg_prd
+                ,'{self.ITERATION}' as itrtn_num
+                ,(select row_cnt from record_count) as tot_rec_cnt
```

Comment:
will it be easier to add an additional join to get the row_cnt from record_count table

Reply:
This is mimicking the SAS code. The `record_count` is a view that just has a count. It's defined on line 285 in this file.

Reply:
It's a good question, though. I suppose this could be changed to `(select count(*) from {self.DA_SCHEMA}.TAF_ANN_UP_{tblname} where da_run_id = {self.DA_RUN_ID}) as tot_rec_cnt` and the `from` part of this query could be removed since it only inserts 1 row. But since we're mostly copying the legacy system's code, deviating too far from that might cause an issue if we need to trace through the legacy code and compare the current code to the legacy code.

Reply:
Does this override the TAF_Runner - create_efts_metadata() for UP? It looks like you have specific SQL for UP so that makes sense.

Reply:
@FICU-tensileai Your comment made me rethink how I was doing this. I ended up going through APR and APL's SAS code to see how things were being done. I moved this part to TAF_Runner.py and I think this new solution can just be applied to APR, APL, and UP. DE is a different story.